GVRf/Framework/src/org/gearvrf/GVRSceneObject.java

```diff
@@ -143,29 +143,76 @@ public GVRSceneObject(GVRContext gvrContext, GVRMesh mesh,
      * that, say, {@code sceneObject2} and {@code sceneObject3} are using the
      * same mesh as {@code sceneObject1}, and will only load the mesh once.
      * 
-     * @param context
+     * @param gvrContext
      *            current {@link GVRContext}.
      * @param mesh
      *            Basically, a stream containing a mesh file.
      * @param texture
      *            Basically, a stream containing a texture file. This can be
      *            either a compressed texture or a regular Android bitmap file.
-     *            
-     * @since 1.6.7            
+     * 
+     * @since 1.6.7
      */
-    public GVRSceneObject(GVRContext context, GVRAndroidResource mesh,
+    public GVRSceneObject(GVRContext gvrContext, GVRAndroidResource mesh,
             GVRAndroidResource texture) {
-        this(context);
+        this(gvrContext);
 
         // Create the render data
-        GVRRenderData renderData = new GVRRenderData(context);
+        GVRRenderData renderData = new GVRRenderData(gvrContext);
+
+        // Set the mesh
+        renderData.setMesh(gvrContext.loadFutureMesh(mesh));
+
+        // Set the texture
+        Future<GVRTexture> futureTexture = gvrContext
+                .loadFutureTexture(texture);
+        GVRMaterial material = new GVRMaterial(gvrContext);
+        material.setMainTexture(futureTexture);
+        renderData.setMaterial(material);
+
+        // Attach the render data
+        attachRenderData(renderData);
+    }
+
+    /**
+     * Very high-level constructor that Constructs a rectangular scene object
+     * (specified by the width and height) and asynchronously loads the texture.
+     * 
+     * Note that because of <a href="package-summary.html#async">asynchronous
+     * request consolidation</a> you generally don't have to do anything special
+     * to create several objects that share the same mesh or texture: if you
+     * create all the objects in {@link GVRScript#onInit(GVRContext) onInit(),}
+     * the meshes and textures will generally <em>not</em> have loaded before
+     * your {@code onInit()} method finishes. Thus, the loading code will see
+     * that, say, {@code sceneObject2} and {@code sceneObject3} are using the
+     * same mesh as {@code sceneObject1}, and will only load the mesh once.
+     * 
+     * @param gvrContext
+     *            current {@link GVRContext}.
+     * @param width
+     *            the scene object's width
+     * @param height
+     *            the scene object's height
+     * @param texture
+     *            Basically, a stream containing a texture file. This can be
+     *            either a compressed texture or a regular Android bitmap file.
+     * 
+     * @since 1.6.7
+     */
+    public GVRSceneObject(GVRContext gvrContext, float width, float height,
```

Comment:
I don't think that this is really the right approach: we have a lot of code that's _almost_ the same as the constructor above, and we're adding a `Future<>` version of just one of the many constructor permutations.

I think it would probably be better to refactor the above `(GVRContext, GVRAndroidResource, GVRAndroidResource)` constructor:

    // almost all the existing logic
    public GVRSceneObject(GVRContext, Future<GVRMesh>, Future<GVRTexture>);
    // calls above
    public GVRSceneObject(GVRContext, GVRAndroidResource, GVRAndroidResource);

... then we can add some trivial methods to GVRContext:

    Future<GVRMesh> futureMesh(GVRMesh);
    Future<GVRTexture> futureMesh(GVRTexture);

(I can write those two, if you like.)